src/github.com/getlantern/detour/direct_conn.go

```diff
@@ -0,0 +1,144 @@
+package detour
+
+import (
+	"fmt"
+	"net"
+	"sync/atomic"
+)
+
+type directConn struct {
+	net.Conn
+	addr string
+	// keep track of the total bytes read by this connection, atomic
+	readBytes uint64
+	// a flag telling if connection is closed
+	closed uint32
+}
+
+var (
+	blockDetector atomic.Value
+)
+
+// SetCountry sets the ISO 3166-1 alpha-2 country code
+// to load country specific detection rules
+func SetCountry(country string) {
+	blockDetector.Store(detectorByCountry(country))
+}
+
+func init() {
+	blockDetector.Store(detectorByCountry(""))
+}
+
+func dialDirect(network string, addr string, ch chan conn) {
+	go func() {
+		log.Tracef("Dialing direct connection to %s", addr)
+		conn, err := net.DialTimeout(network, addr, TimeoutToConnect)
+		detector := blockDetector.Load().(*Detector)
+		if err == nil {
+			if detector.DNSPoisoned(conn) {
+				_ = conn.Close()
```

Comment:
Just want to ignore the errors when closing `conn`. Add a debug message would be better. Thanks for pointing out.